File: BubbleSort.c
#include<stdio.h>
void main()
{
	int i,j,no,c,temp;
	printf(" Enter The Size of 1D Array\n");
	scanf(" %d", &no);
	int arr[no];
	printf(" Enter The Elements into the Array: \n");
	for(i=0;i<no;i++)
	scanf("%d",&arr[i]);
	printf("\n\n The Given Array Is:\n \n");
	for(i=0;i<no;i++)
	printf("%d ",arr[i]);
	/*BUBBLE SORT*/
	printf("\n\n The Sorted Array Is: \n\n");
	for(j=0;j<no;j++)
	{
		for(i=(no-1);i>c;i--)
		{
			if(arr[i]<arr[i-1])
			{
				temp=arr[i-1];
				arr[i-1]=arr[i];
				arr[i]=temp;
			}
			else
			continue;
		}
		++c;
	}
	for(i=0;i<no;i++)
	printf("%d ", arr[i]);
	printf("\n");
}	
			

File: Sorting/BubbleSort.java
class BubbleSort{
	public void BubbleSortRoutine(int n, int arr[]){
		int i=0,j=0;
		for(i=1;i<n-1;i++)
			for(j=0;j<n-1-1;j++)
				if(arr[j+1]<arr[j])
					Swap(i,j,arr);
	}
	public void Swap(int i, int j, int arr[]){
		int temp = arr[i];
		arr[i]=arr[j];
		arr[j]=temp;
	}
}		
	